src/com/rabbitmq/client/impl/recovery/AutorecoveringConnection.java

```diff
@@ -428,18 +470,43 @@ private void recoverBlockedListeners() {
         }
     }
 
-    private void recoverConnection() throws IOException, InterruptedException {
-        boolean recovering = true;
-        while (recovering) {
+	//Returns true if the connection was recovered, 
+	//false if application initiated shutdown while attempting recovery.  
+    private boolean recoverConnection() throws IOException, InterruptedException {
+        while (!manuallyClosed) //manuallyClosed marked volatile -ensures thread cannot cache value of manuallyClosed.
```

Comment:
If the intent of using `while` here is to bypass thread-local caching, this is redundant since the field is volatile. If the intent is to double check field value, this is quite unclear. I'd rather move connection initiation inside the synchronised block (we want to only ever have 1 delegate per autorecovering connection anyway).